src/users/users.controller.ts

```diff
@@ -54,6 +53,18 @@ export class UsersController {
     private caslAbilityFactory: CaslAbilityFactory,
   ) {}
 
+  async generateDatasetInstanceForPermissions(
+    user: JWTUser,
+    pid?: string,
+  ): Promise<DatasetClass> {
+    const datasetInstance = new DatasetClass();
+    datasetInstance._id = "";
+    datasetInstance.pid = pid || "";
+    datasetInstance.accessGroups = user.currentGroups || [];
+    datasetInstance.ownerGroup = user.currentGroups[0] || "";
+
+    return datasetInstance;
+  }
```

Comment:
Why is this function here? Should it belong to the dataset sub-system?

Reply:
I forgot to remove this. Previously it was intended to check what kind of dataset users can create - withpid, or without pid for `/:id/authorization/dataset/create`.  But after discussion we decided to return only true or false of whether user can create dataset, since the boolean value will be used on Frontend  to display `CreateDataSet` button

Reply:
Great. Thank for clarifying